### PR TITLE
CMR-8046 - a bug fix to isolate parse errors which can happen if CSV …

### PR DESCRIPTION
**Pull request by jceaser**
Background: If KMS changes it's structure and CMR team does not get notified, CMR will fail to parse the new CSV file. When this happens, CMR use to throw an exception from inside a for loop where CMR integrates over schema names (platforms, instruments, science keywords) to download CSV files to process. When this happens nothing catches the exception and the for loop stops processing schemes meaning all schemes with valid data which happens latter in the loop do not get processed.

Fix: Don't throw exception. Instead, return true/false in a tester function and handle that state latter on. I also added some tests since we need to increase code coverage anyways.

### CODE REVIEW

File: transmit-lib/src/cmr/transmit/kms.clj [hunk 1]
```diff
@@ -168,42 +168,45 @@
 
 (defn- validate-subfield-names
```
Comment (ygliuvt) [inline transmit-lib/src/cmr/transmit/kms.clj:169]: change the function name to something like `subfield-names-valid?`

**Review by ygliuvt — APPROVED** [review]

**Review by eereiter — APPROVED** [review]

File: transmit-lib/test/cmr/transmit/test/kms.clj [hunk 2]
```diff
@@ -77,13 +77,57 @@
           actual (#'cmr.transmit.kms/parse-entries-from-csv :mime-type sample-csv-mimetype)]
       (is (= expected actual))))
 
-  (testing "Invalid subfield names in the CSV throws an exception"
-    (is (thrown-with-msg?
-          Exception
-          #"Expected subfield names for instruments to be"
-          (#'cmr.transmit.kms/parse-entries-from-csv :instruments sample-csv)))))
+  (testing "Invalid subfield names in the CSV returns nothing"
+    (is nil? (#'cmr.transmit.kms/parse-entries-from-csv :platforms sample-csv))))
```
Comment (jaybarra) [inline transmit-lib/test/cmr/transmit/test/kms.clj:81]:
```suggestion
  (testing "Invalid subfield names in the CSV returns nothing"
    (is (nil? (#'cmr.transmit.kms/parse-entries-from-csv :platforms sample-csv)))))
```
```

**Review by jaybarra — CHANGES_REQUESTED** [review]

File: transmit-lib/test/cmr/transmit/test/kms.clj [hunk 3]
```diff
@@ -77,13 +77,57 @@
...
+    (cmr.transmit.config/set-kms-scheme-override-json! "{}")
+    (let [result (#'cmr.transmit.kms/scheme-overrides)]
+      (is (= {} result))))
+
+  (testing "No configuration value should return nil"
+    ;; doing this test last so that the setting goes back to the default for
+    ;; tests that follow this call.
+    (cmr.transmit.config/set-kms-scheme-override-json! "")
+    (let [result (#'cmr.transmit.kms/scheme-overrides)]
+      (is (= nil result)))))
```
Comment (jaybarra) [inline transmit-lib/test/cmr/transmit/test/kms.clj:119]:
```suggestion
      (is (nil? result)))))
```

File: transmit-lib/test/cmr/transmit/test/kms.clj [hunk 4]
```diff
@@ -77,13 +77,57 @@
...
+
```
Comment (jaybarra) [inline transmit-lib/test/cmr/transmit/test/kms.clj:120]: Whitespace, extra newline

Comment (jaybarra) [inline transmit-lib/test/cmr/transmit/test/kms.clj:120]:
```suggestion
```

File: transmit-lib/test/cmr/transmit/test/kms.clj (hunk 3 above)
Comment (jceaser) [reply transmit-lib/test/cmr/transmit/test/kms.clj:119]: I rewrote this section to use are3

File: transmit-lib/test/cmr/transmit/test/kms.clj (hunk 2 above)
Comment (jceaser) [reply transmit-lib/test/cmr/transmit/test/kms.clj:81]: thanks

**Review by jaybarra — APPROVED** [review]

**Pull request merged**